src/server/work_runner.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::{Context, Ok, Result};
+use async_trait::async_trait;
+use bb8_redis::{
+    bb8::{Pool, PooledConnection},
+    redis::AsyncCommands,
+    RedisConnectionManager,
+};
+use workers::Work;
+
+use crate::{
+    msg::{Message, MessageBuilder},
+    util::Queue,
+};
+
+use super::{HandlerInput, HandlerOutput, Module};
+
+type Reply = MessageBuilder;
+
+#[derive(Clone)]
+pub struct WorkRunner {
+    pool: Pool<RedisConnectionManager>,
+}
+
+impl WorkRunner {
+    pub fn new(pool: Pool<RedisConnectionManager>) -> Self {
+        WorkRunner { pool }
+    }
+
+    async fn get_connection(&self) -> Result<PooledConnection<'_, RedisConnectionManager>> {
+        let conn = self
+            .pool
+            .get()
+            .await
+            .context("unable to retrieve a redis connection from the pool")?;
+
+        Ok(conn)
+    }
+
+    async fn prepare(msg: &mut Message, result: HandlerOutput) {
+        msg.data = base64::encode(result.data);
+
+        let src = msg.source;
+        msg.source = msg.destination[0];
+        msg.destination = vec![src];
+    }
+
+    async fn send(&self, msg: Message) -> Result<()> {
+        let mut conn = self.get_connection().await?;
+        conn.rpush(Queue::Reply.as_ref(), msg)
+            .await
+            .context("unable to send your reply message")?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Work for WorkRunner {
+    type Input = (String, Message, &'static Module);
```

Comment:
This (&'static Module) will not be possible in practice. I think the right way to do it is that if your WorkRunner just takes ownership of the `root` module from the server (when run is called). 

This way the Input type can only be `(String, Message)` the worker then can lookup the handler. 
Execute it and process the returned data.

So I think what needs to be done is 
- on the server `run(self)`
  - create the pool, the worker runner takes ownership of the root module.
  - WorkerRunner doesn't need to be clonable because u already wrap it in an Arc (which is good in our case because a Module is not clonable)
  -